azurelinuxagent/distro/default/env.py

```diff
@@ -94,6 +100,22 @@ def handle_dhclient_restart(self):
            self.distro.dhcp_handler.conf_routes()
            self.dhcpid = newpid
 
+     def handle_fstab_update(self):
+        # Only try to mount if the fstab has been modified. This avoids background mounts when admins are working on a unmounted partition
+        # If we find an error, backoff and only check every minute to give time to correct it and avoid filling the log
+        fstabCurrentModifiedStamp = shellutil.run_get_output("stat /etc/fstab -c %Y")
+        if (fstabCurrentModifiedStamp != self.fstabModifiedStamp) and (datetime.datetime.now() > self.lastNotice + datetime.timedelta(seconds=60)):
+            ret, output = shellutil.run_get_output("mount -av")
```

Comment:
I evaluated it initially, but sadly `mount --fake` doesn't do the level of validation that I believe is necessary to catch certain conditions.  I try to illustrate this in the attached script. If you run `fstabTests.sh` and `fstabTests.sh "-f"` you will see that several of the scenarios are not caught.
[fstabTests.txt](https://github.com/Azure/WALinuxAgent/files/314265/fstabTests.txt)

In the scenario you describe above, simply doing an `umount` shouldn't cause it to remount as the `mount` code is only triggered when fstab is modified.  That said, if an admin `umount` _and_ changed fstab then that would indeed cause the unmounted partition to be mounted.  However, if the fstab is valid this should be a 1-time occurrence.  It seems like an edge case with minimal impact but I am open to feedback on this if folks feel differently.  This also goes to the principal in the [1-pager](https://github.com/Azure/WALinuxAgent/files/305715/fstab.checking_public_1-pager.docx) of 
>"A VM could rebooted at any time, hence the fstab file should be valid at all times."